docs/src/main/tut/entity.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: default
+title: Dealing with Entity Bodies
+---
+
+## Why Entity*
+
+Http4s handles HTTP requests and responses in a streaming fashion. Your service
+will receive a request after the header has been parsed (ok, not 100%
+streaming), but before the body has been fully received. The same applies to the
+http client usage, where you can start a connection before the body is fully
+materialized. You don't have to load the full body into memory to submit the
+request either. Taking a look at `Request` and `Response`, both have a body of
+type `EntityBody`, which is simply an alias to `Process[Task, ByteVector]`. To
+understand `Process`, take a look at the [streams-tutorial].
+
+The `EntityDecoder` and `EntityEncoder` help with the streaming nature of the
+data in a http body, and they also have additional logic to deal with media
+types. Not all decoders are streaming, depending on the implementation.
+
+## Working with Media Types
```

Comment:
This paragraph seems a bit scattered to me: it talks about EntityDecoder, and then the headers that EntityEncoder adds, and then back to how to do an EntityDecoder.  I would cover one, then the other.